src/coldfront_plugin_api/utils.py

```diff
@@ -0,0 +1,50 @@
+from coldfront.core.user import utils as user_utils
+from django.contrib.auth.models import User
+from django.core.exceptions import ObjectDoesNotExist
+
+
+def find_user(username):
+    """Searches for a user in the configured search providers."""
+    search = user_utils.CombinedUserSearch(username, "username_only")
+    results = search.search().get("matches")
+
+    found = None
+    if results:
+        for entry in results:
+            if entry.get("username") == username:
+                found = entry
+
+    return found
+
+
+def create_user(username, first_name, last_name, email):
+    user_obj, _ = User.objects.get_or_create(
+        username=username)
```

Comment:
Will this make unintentional changes if `username` exists but with a different `first_name`/`last_name`/`email`?

Reply:
the `get_or_create` method will not. The rest of this method (`create_user`) will. It is not unlike the process of authenticating, which will sync up the first_name, last_name, email upon each authentication of a user.